src/zkp/ecdsa_adaptor.rs

```diff
@@ -198,17 +197,16 @@ impl EcdsaAdaptorSignature {
         let mut adaptor_sig = ffi::EcdsaAdaptorSignature::new();
 
         unsafe {
-            debug_assert!(
-                ffi::secp256k1_ecdsa_adaptor_encrypt(
-                    *secp.ctx(),
-                    &mut adaptor_sig,
-                    sk.as_c_ptr(),
-                    enckey.as_c_ptr(),
-                    msg.as_c_ptr(),
-                    ffi::secp256k1_nonce_function_ecdsa_adaptor,
-                    aux_rand.as_c_ptr() as *mut ffi::types::c_void,
-                ) == 1
+            let res = ffi::secp256k1_ecdsa_adaptor_encrypt(
+                *secp.ctx(),
+                &mut adaptor_sig,
+                sk.as_c_ptr(),
+                enckey.as_c_ptr(),
+                msg.as_c_ptr(),
+                ffi::secp256k1_nonce_function_ecdsa_adaptor,
+                aux_rand.as_c_ptr() as *mut ffi::types::c_void,
             );
+            debug_assert_eq!(res, 1);
```

Comment:
To be consistent with other files in this repo, can we have the `unsafe` block evaluate to `res`?

That will also keep the unsafe as short as possible.